RxJavaTemplate/buildSrc/src/main/java/Dependencies.kt

```diff
@@ -0,0 +1,127 @@
+object AppModule {
+    val implementations = listOf(
+        KOTLIN_STDLIB_JDK7,
+        KOTLIN_STDLIB,
+
+        ANDROID_MATERIAL,
+        ANDROIDX_CORE,
+        ANDROIDX_FRAGMENT_KTX,
+        ANDROIDX_LEGACY_SUPPORT,
+        ANDROIDX_APPCOMPAT,
+        ANDROIDX_RECYCLERVIEW,
+        ANDROIDX_CONSTRAINT_LAYOUT,
+        ANDROIDX_MULTIDEX,
+
+        ANDROIDX_NAVIGATION_RUNTIME_KTX,
+        ANDROIDX_NAVIGATION_FRAGMENT_KTX,
+        ANDROIDX_NAVIGATION_UI_KTX,
+
+        ANDROIDX_LIFECYCLE_VIEWMODEL_KTX,
+        ANDROIDX_LIFECYCLE_EXTENSIONS,
+        ANDROIDX_LIFECYCLE_COMMON_JAVA8,
+        ANDROIDX_LIFECYCLE_RUNTIME,
+
+        RX_BINDING,
+        RX_BINDING_CORE,
+
+        HILT,
+        TIMBER,
+        GLIDE,
+        RX_PERMISSION
+    )
+
+    val debugImplementations = listOf(
+        ANDROIDX_FRAGMENT_TESTING,
+        CHUCKER
+    )
+
+    val releaseImplementations = listOf(
+        CHUCKER_NO_OP
+    )
+
+    val kapts = listOf(
+        ANDROIDX_LIFECYCLE_COMPILER,
+        GLIDE_COMPILER,
+        HILT_COMPILER
+    )
+
+    val androidTestKapts = listOf(
+        HILT_ANDROID_COMPILER
+    )
+
+    val androidTestImplementations = listOf(
+        ANDROIDX_TEST_RUNNER,
+        ANDROIDX_TEST_RULES,
+
+        ANDROIDX_TEST_ESPRESSO_CORE,
+        ANDROIDX_TEST_ESPRESSO_INTENTS,
+        ANDROIDX_TEST_ESPRESSO_CONTRIB,
+        ANDROIDX_TEST_ESPRESSO_WEB,
+
+        HILT_ANDROID_TESTING,
+        MOCKITO_INLINE
+    )
+
+    val testImplementations = listOf(
+        JUNIT,
+        KLUENT_ANDROID,
+        MOCKITO_KOTLIN,
+        ROBOLECTRIC
+    )
+}
+
+object DataModule {
+    val implementations = listOf(
+        KOTLIN_STDLIB_JDK7,
+        KOTLIN_STDLIB,
+        ANDROIDX_CORE
+    )
+
+    val apis = listOf(
+        RETROFIT,
+        RETROFIT_ADAPTER_RXJAVA2,
+        RETROFIT_CONVERTER_MOSHI,
+
+        MOSHI_KOTLIN,
+        MOSHI_ADAPTERS,
+
+        OKHTTP3,
+        OKHTTP3_LOGGING
+    )
+
+    val testImplementations = listOf(
+        JUNIT,
+        MOCKITO_KOTLIN
+    )
+}
+
+object DomainModule {
+    val implementations = listOf(
+        KOTLIN_STDLIB_JDK7,
+        KOTLIN_STDLIB,
+
+        ANDROIDX_CORE,
+        ANDROID_SECURITY_CRYPTO,
+
+        HILT,
+        RX_ANDROID
+    )
+
+    val testImplementations = listOf(
+        JUNIT,
+        KLUENT_ANDROID,
+        MOCKITO_KOTLIN,
+        ROBOLECTRIC
+    )
+}
+
+object CommonRxModule {
+    val implementations = listOf(
+        KOTLIN_STDLIB_JDK7
+    )
+
+    val apis = listOf(
+        RX_JAVA,
+        RX_KOTLIN
+    )
+}
```

Comment:
There are still some concerns from me to manage project dependencies by this approach. Because this change affects too much on the base structure of our template and the way to use it in our development process so I recommend having an open discussion before continuing 🙏

Reply:
I agree